pkg/kubefed2/federate/util.go

```diff
@@ -106,24 +108,36 @@ func LookupAPIResource(config *rest.Config, key, targetVersion string) (*metav1.
 				lowerKey == strings.ToLower(resource.Kind) ||
 				lowerKey == fmt.Sprintf("%s.%s", resource.Name, gv.Group) {
 
-				targetResource = &resource
-				break
+				if targetResource != nil {
```

Comment:
(No action required) Consider creating an anonymous function that can be reused across both match points.

Reply:
addressed